pkg/apis/pipeline/v1/container_types.go

```diff
@@ -541,3 +570,32 @@ func (s *Sidecar) SetContainerFields(c corev1.Container) {
 	s.StdinOnce = c.StdinOnce
 	s.TTY = c.TTY
 }
+
+// GetVarSubstitutionExpressions walks all the places a substitution reference can be used
+func (s *Sidecar) GetVarSubstitutionExpressions() ([]string, bool) {
+	var allExpressions []string
+	allExpressions = append(allExpressions, validateString(s.Name)...)
+	allExpressions = append(allExpressions, validateString(s.Image)...)
+	allExpressions = append(allExpressions, validateString(string(s.ImagePullPolicy))...)
+	allExpressions = append(allExpressions, validateString(s.Script)...)
+	for _, cmd := range s.Command {
+		allExpressions = append(allExpressions, validateString(cmd)...)
+	}
+	for _, arg := range s.Args {
+		allExpressions = append(allExpressions, validateString(arg)...)
+	}
+	for _, env := range s.Env {
+		allExpressions = append(allExpressions, validateString(env.Value)...)
+		if env.ValueFrom != nil {
+			if env.ValueFrom.SecretKeyRef != nil {
+				allExpressions = append(allExpressions, validateString(env.ValueFrom.SecretKeyRef.Key)...)
+				allExpressions = append(allExpressions, validateString(env.ValueFrom.SecretKeyRef.LocalObjectReference.Name)...)
+			}
+			if env.ValueFrom.ConfigMapKeyRef != nil {
+				allExpressions = append(allExpressions, validateString(env.ValueFrom.ConfigMapKeyRef.Key)...)
+				allExpressions = append(allExpressions, validateString(env.ValueFrom.ConfigMapKeyRef.LocalObjectReference.Name)...)
+			}
+		}
+	}
+	return allExpressions, len(allExpressions) != 0
+}
```

Comment:
if this function does not yet exist, does this mean we don't support variable replacements in sidecars?

Reply:
hi @jerop , we support parameter substitution replacements in containers and sidecars
the specific implementation function can be seen here:
https://github.com/tektoncd/pipeline/blob/5dd04bf5578b38ed6640b44849e8b9b65a3900e5/pkg/container/container_replacements.go#L26-L48

parameter substitution replacements in sidecar and contianer are obtained from params. They do not directly replace ${{ tasks.xxx.results.xxx }}, but are implemented in the form of ${{ params.xxx }}.

I added this function to obtain the `${{ tasks.xxx.results.xxx }}` subexpressions in the container and sidecar, so that the reference of the task result can enter the Dag judgment, and let the tasks that refer to other results be executed afterwards.

https://github.com/tektoncd/pipeline/blob/5dd04bf5578b38ed6640b44849e8b9b65a3900e5/pkg/apis/pipeline/v1/resultref.go#L171-L184

As stated in the comments on this issue
https://github.com/tektoncd/pipeline/issues/7086#issuecomment-1711191897

Reply:
sounds great, thanks for the clarifications!